sql/plan/indexed_join.go

```diff
@@ -31,13 +27,11 @@ func (ij *IndexedJoin) JoinType() JoinType {
 	return ij.joinType
 }
 
-func NewIndexedJoin(primaryTable, indexedTable sql.Node, joinType JoinType, cond sql.Expression, primaryTableExpr []sql.Expression, index sql.Index) *IndexedJoin {
+func NewIndexedJoin(left, right sql.Node, joinType JoinType, cond sql.Expression) *IndexedJoin {
```

Comment:
It can't implement Expressioner, because the expressions it contains refer to columns passed in through the RowIter function, not coming from child nodes.  So the analyzer would complain about not being able to find columns. I might try to formalize this relationship in a future pass, but it's special to these nodes at the moment.